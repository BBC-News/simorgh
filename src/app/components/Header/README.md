# Header [![Storybook](https://github.com/storybooks/press/blob/master/badges/storybook.svg)](https://simorghstorybook.now.sh/?selectedKind=Header)

### Description
This component functions as a wrapper for the [Brand](../Brand) component (which is also used in the [Footer](../Footer)). The wrapper is a HTML 'header' element with the 'role' of 'banner' for accessibility. This component is distinct from Brand as the 'header' HTML element is unique and cannot be reused elsewhere on the page.
	
### When to use this component
Header is designed to be used at the top of the document above all other content.

As this component uses the [Brand](../Brand), which consumes the service context via the [ServiceContextConsumer](../ServiceContext) component, the service context must be provided to the Header by the [ServiceContextProvider](../ServiceContext).

### Accessibility notes
Landmark:
Banner
