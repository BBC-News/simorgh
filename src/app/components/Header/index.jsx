import React from 'react';
import Brand from '../Brand';

const Header = () => (
  <header role="banner">
    <Brand />
  </header>
);

export default Header;
