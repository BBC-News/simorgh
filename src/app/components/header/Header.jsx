import React, { Component, Fragment } from 'react';
import styled from 'styled-components';

const BBCNewsBranding = styled.div`
  background-color: #bb1919;
  height: 58px;
  width: 100%;
`;

const BBCNewsBrandingLogo = styled.div`
  margin: 0 62px 0 62px;
  padding: 10px 16px 10px 16px;
`;

const BBCNewsNavigation = styled.div`
  width: 100%;
`;

const BBCNewsLink = styled.a`
  font-family: ReithSans, Arial, Helvetica, freesans, sans-serif;
  font-size: 2.4em;
  color: #fff;
  text-transform: uppercase;
`;

class Header extends Component {
  state = {
    title: 'BBC News',
  };

  render() {
    const { title } = this.state;

    return (
      <Fragment>
        <BBCNewsBranding>
          <BBCNewsBrandingLogo>
            <BBCNewsNavigation>
                  <BBCNewsLink href="/news" id="brand">
                    {title}
                  </BBCNewsLink>
            </BBCNewsNavigation>
          </BBCNewsBrandingLogo>
        </BBCNewsBranding>
      </Fragment>
    );
  }
}

export default Header;