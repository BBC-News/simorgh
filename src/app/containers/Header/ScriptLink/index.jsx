import React, { useContext } from 'react';
import { compile } from 'path-to-regexp';
import { useRouteMatch } from 'react-router-dom';
import ScriptLink from '@bbc/psammead-script-link';
import VisuallyHiddenText from '@bbc/psammead-visually-hidden-text';
import { UserContext } from '#contexts/UserContext';
import { ServiceContext } from '#contexts/ServiceContext';

export const getVariantHref = ({ path, params, variant }) =>
  compile(path)(
    {
      ...params,
      variant: `/${variant}`,
      amp: undefined, // we don't want to link to AMP pages directly
    },
    {
      encode: value => value,
    },
  );

const ScriptLinkContainer = () => {
  const { setPreferredVariantCookie } = useContext(UserContext);
  const { service, script, scriptLink } = useContext(ServiceContext);

  const { text, offscreenText, variant } = scriptLink;
  const { path, params } = useRouteMatch();

  return (
    <ScriptLink
      script={script}
      service={service}
      href={getVariantHref({
        path,
        params,
        service,
        variant,
      })}
      variant={variant}
      onClick={() => setPreferredVariantCookie(service, variant)}
    >
      <span aria-hidden="true">{text}</span>
      <VisuallyHiddenText>{` ${offscreenText}`}</VisuallyHiddenText>
    </ScriptLink>
  );
};

export default ScriptLinkContainer;
