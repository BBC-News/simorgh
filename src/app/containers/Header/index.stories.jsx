import React from 'react';
import { storiesOf } from '@storybook/react';
import { inputProvider } from '@bbc/psammead-storybook-helpers';
import { withKnobs } from '@storybook/addon-knobs';
import HeaderContainer from '.';
import { RequestContextProvider } from '#contexts/RequestContext';
import services from '#testHelpers/serviceConfigs';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContextProvider } from '#contexts/ToggleContext';

storiesOf('Containers|Header', module)
  .addParameters({ chromatic: { disable: true } })
  .addDecorator(withKnobs)
  .add(
    'default',
    inputProvider({
      // eslint-disable-next-line react/prop-types
      componentFunction: ({ service }) => {
        return (
          <ToggleContextProvider>
            <ServiceContextProvider service={service}>
              <RequestContextProvider
                isAmp={false}
                pageType="frontPage"
                service={service}
                bbcOrigin="https://www.test.bbc.com"
              >
                <HeaderContainer />
              </RequestContextProvider>
            </ServiceContextProvider>
          </ToggleContextProvider>
        );
      },
      services: Object.keys(services),
    }),
  );
