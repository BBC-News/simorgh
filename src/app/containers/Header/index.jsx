import React, { useContext } from 'react';
import SkipLink from '@bbc/psammead-brand/skip-link';
import BrandContainer from '../Brand';
import NavigationContainer from '../Navigation';
import { ServiceContext } from '#contexts/ServiceContext';
import { RequestContext } from '#contexts/RequestContext';
import ConsentBanner from '../ConsentBanner';
import ScriptLink from './ScriptLink';
import useToggle from '#hooks/useToggle';
import onClient from '#lib/utilities/onClient';

const HeaderContainer = () => {
  const { pageType } = useContext(RequestContext);
  const { service, script, translations, dir, scriptLink } = useContext(
    ServiceContext,
  );
  const { skipLinkText } = translations;
  const borderBottom = pageType !== 'frontPage';

  // The article page toggles the nav bar based on environment
  const showNavOnArticles = useToggle('navOnArticles').enabled;

  // All other page types show the nav bar at all times
  const showNav = showNavOnArticles || pageType !== 'article';

  const isOperaMini = onClient() && window.operamini;

  const skipLink = !isOperaMini && (
    <SkipLink service={service} script={script} dir={dir} href="#content">
      {skipLinkText}
    </SkipLink>
  );

  return (
    <header role="banner">
      <ConsentBanner />
      <BrandContainer
        borderBottom={borderBottom}
        skipLink={skipLink}
        scriptLink={scriptLink && <ScriptLink />}
      />
      {showNav && <NavigationContainer />}
    </header>
  );
};

export default HeaderContainer;
