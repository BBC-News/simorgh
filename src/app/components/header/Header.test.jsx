import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

describe('Header', () => {
  describe('Component', () => {
    it('should render correctly', () => {
      const tree = renderer.create(<Header />);
      expect(tree).toMatchSnapshot();
    });
  });
});
