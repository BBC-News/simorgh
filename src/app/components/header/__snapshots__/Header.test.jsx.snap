// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
<header
  aria-label="BBC News"
  className="sc-bwzfXH fEmsWu"
  role="banner"
>
  <a
    className="sc-bdVaJa iguvfJ"
    href="/news"
    id="brand"
  >
    BBC News
  </a>
</header>
`;
